frontend/src/components/administration/users/dialog-edit-user.tsx

```diff
@@ -0,0 +1,73 @@
+import { FC, useEffect, useState } from 'react';
+
+import { User, Role } from '@autoline/shared/common/types/types';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import FormControl from '@mui/material/FormControl';
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import OutlinedInput from '@mui/material/OutlinedInput';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+
+type DialogEditUserProps = {
+  user?: User;
+  open: boolean;
+  handleClose: () => void;
+  handleSubmit: (user: User, newRole: Role) => void;
+};
+
+const DialogEditUser: FC<DialogEditUserProps> = (props) => {
+  const { user, open, handleClose, handleSubmit } = props;
+  const [role, setRole] = useState<Role | undefined>(user?.role);
+
+  useEffect(() => {
+    if (user && open) {
+      setRole(user.role);
+    }
+  }, [user, open]);
```

Comment:
```suggestion
  useEffect(() => {
    if (user?.role && open) {
      setRole(user.role);
    }
  }, [user?.role, open]);
```
In such way minimize the rerenders, because we react to the primitive value instead of the object

Reply:
Agree, thanks.